moveit_core/CMakeLists.txt

```diff
@@ -57,46 +56,49 @@ find_package(visualization_msgs REQUIRED)
 # TODO: Port python bindings
 # find_package(pybind11 REQUIRED)
 
-# Finds Boost Components
-include(ConfigExtras.cmake)
-
-add_subdirectory(collision_detection_bullet)
-add_subdirectory(collision_detection_fcl)
-add_subdirectory(collision_detection)
-add_subdirectory(collision_distance_field)
-add_subdirectory(constraint_samplers)
-add_subdirectory(controller_manager)
-add_subdirectory(distance_field)
-add_subdirectory(dynamics_solver)
-add_subdirectory(exceptions)
-add_subdirectory(kinematic_constraints)
-add_subdirectory(kinematics_base)
-add_subdirectory(kinematics_metrics)
-add_subdirectory(macros)
-add_subdirectory(online_signal_smoothing)
-add_subdirectory(planning_interface)
-add_subdirectory(planning_request_adapter)
-add_subdirectory(planning_scene)
-add_subdirectory(robot_model)
-add_subdirectory(robot_state)
-add_subdirectory(robot_trajectory)
-add_subdirectory(trajectory_processing)
-add_subdirectory(transforms)
-add_subdirectory(utils)
-add_subdirectory(version)
+include(ConfigExtras.cmake) # Finds Boost Components
+include(GenerateExportHeader)
+
+add_library(moveit_core SHARED)
+add_library(moveit::core ALIAS moveit_core)
+generate_export_header(moveit_core EXPORT_FILE_NAME include/moveit/export.h)
+target_include_directories(moveit_core PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
```

Comment:
I'm not convinced we gain anything from splitting sources and headers for the different components. I'll not block the decision though if you think it more convenient.

The decision to split the headers of is fully independent of whether or not there should be one `libmoveit_core.so` and it's definitely not considered separately in this discussion so far.
Yes, it consolidates paths a bit. But it also adds three subdirectory changes to go from a component's sources to the respective headers...